kernel/src/process.rs

```diff
@@ -431,6 +441,13 @@ pub trait ProcessType {
     /// Increment the number of times the process called a syscall and record
     /// the last syscall that was called.
     fn debug_syscall_called(&self, last_syscall: Syscall);
+
+    /// Reset some debug information after the process restarts.
+    fn debug_restarted(&self);
+
+    /// Collect information about context switches. Currently only counts
+    /// the number of context switches due to timeslice expirations.
+    fn debug_context_switch(&self, switch_reason: &Option<syscall::ContextSwitchReason>);
```

Comment:
Many of these do not need to be in the trait. If they aren't called from outside process.rs we can just implement them on the struct.

Reply:
Done. Thanks for pointing this out, this lead me to a rabbit hole of compiler settings :) Please see my comment above.